File: 848.shifting-letters.cpp
/*
 * @lc app=leetcode id=848 lang=cpp
 *
 * [848] Shifting Letters
 */
#include <string>
#include <vector>
using namespace std;
// @lc code=start
class Solution
{
public:
    string shiftingLetters(string s, vector<int> &shifts)
    {
        int len = s.size();
        int shift = 0;
        for (int i = len - 1; i != -1; --i)
        {
            shift += shifts[i];
            shift %= 26;
            s[i] = ((s[i] - 'a') + shift) % 26 + 'a';
        }
        return s;
    }
};
// @lc code=end

File: 848.shifting-letters.java
/*
 * @lc app=leetcode id=848 lang=java
 *
 * [848] Shifting Letters
 */

// @lc code=start
class Solution {
    public String shiftingLetters(String s, int[] shifts) {

        int len = s.length();
        StringBuilder s2 = new StringBuilder(len);
        s2.setLength(len);
        int shift = 0;
        for (int i = len - 1; i != -1; --i) {
            shift += shifts[i];
            shift %= 26;
            s2.setCharAt(i, (char) (((s.charAt(i) - 'a') + shift) % 26 + 'a'));
        }
        return s2.toString();
    }
}
// @lc code=end
